improve the documentation on accessors (#4657)
pydata/xarray: doc/internals.rst

.. literalinclude:: examples/_code/accessor_example.py

> In general, the only requirement is that the accessor's ``__init__`` only has a single

Reviewer: ```suggestion
In general, the accessor's ``__init__`` must only have a single
```

> In general, the only requirement is that the accessor's ``__init__`` only has a single
> parameter except from ``self``: the ``Dataset`` or ``DataArray`` object it is supposed
> to work on.

Author: thanks for the hint, @mathause, I didn't notice that when writing it. How about this:
```suggestion
In general, the only restriction on the accessor class is that the ``__init__`` method
has just a single parameter: the ``Dataset`` or ``DataArray`` object it is supposed
to work on.
```

Author: Including @dcherian's suggestion, maybe something like this:
```suggestion
In general, the only restriction on the accessor class is that the ``__init__`` method
must have a single parameter: the ``Dataset`` or ``DataArray`` object it is supposed
to work on.
```

Reviewer: The second one looks good.
